Tasks/DeployAzureResourceGroup/Utility.ps1

```diff
@@ -896,25 +923,28 @@ function Add-AzureVMCustomScriptExtension
     try
     {
         $isExtensionExists = Is-WinRMCustomScriptExtensionExists -resourceGroupName $resourceGroupName -vmName $vmName -extensionName $extensionName
-    Write-Verbose -Verbose "IsExtensionExists: $isExtensionExists"
+        Write-Verbose -Verbose "IsExtensionExists: $isExtensionExists"
 	
-    if($isExtensionExists)
-    {
-        Write-Verbose -Verbose "Skipping the addition of custom script extension '$extensionName' as it already exists"
-        return
-    }
+        if($isExtensionExists)
+        {
+            Write-Verbose -Verbose "Skipping the addition of custom script extension '$extensionName' as it already exists"
+
+            Add-WinRMHttpsNetworkSecurityRuleConfig -resourceGroupName $resourceGroupName -vmId $vmId -ruleName $ruleName -rulePriotity $rulePriotity -winrmHttpsPort $winrmHttpsPort
```

Comment:
I am missing this part. If extension exists, why do you call securityruleconfig? Can't the existing extension be for machinecustomscript?